frappe/desk/doctype/dashboard_chart/dashboard_chart.py

```diff
@@ -192,7 +193,10 @@ def get_chart_config(chart, filters, timespan, timegrain, from_date, to_date):
 		ignore_ifnull = True
 	)
 
-	result = get_result(data, timegrain, from_date, to_date)
+	if chart.chart_type == 'Average':
+		result = get_avg_result(data, timegrain, from_date, to_date)
+	else:
+		result = get_result(data, timegrain, from_date, to_date)
```

Comment:
I don't think we need a separate function for this. In `get_result` we get the total of each timegrain (ex. Monthly). We can add a counter for this and divide by that in `get_result` itself. This will add just a couple of lines of code and avoid code duplication since both functions do almost the same thing.

Also I tested the result, `get_avg_result` doesn't seem to be returning correct values.